tests/qchem/vibrational_tests/test_bosonic.py

```diff
@@ -0,0 +1,1230 @@
+        """Test that a BoseWord can be multiplied onto a number (from the left)
+        and return a BoseSentence"""
+        assert bw * number == result
+
+    @pytest.mark.parametrize("bw, number, result", WORDS_AND_NUMBERS_MUL)
+    def test_rmul_number(self, bw, number, result):
+        """Test that a BoseWord can be multiplied onto a number (from the right)
+        and return a BoseSentence"""
+        assert number * bw == result
+
+    tup_bw_mult_error = ((bw4, "string"),)
+
+    @pytest.mark.parametrize("bw, bad_type", tup_bw_mult_error)
+    def test_mul_error(self, bw, bad_type):
+        """Test multiply with unsupported type raises an error"""
+        with pytest.raises(TypeError, match=f"Cannot multiply BoseWord by {type(bad_type)}."):
+            bw * bad_type  # pylint: disable=pointless-statement
+
+    @pytest.mark.parametrize("bw, bad_type", tup_bw_mult_error)
+    def test_rmul_error(self, bw, bad_type):
+        """Test __rmul__ with unsupported type raises an error"""
+        with pytest.raises(TypeError, match=f"Cannot multiply BoseWord by {type(bad_type)}."):
+            bad_type * bw  # pylint: disable=pointless-statement
+
+    @pytest.mark.parametrize("bw, bad_type", tup_bw_mult_error)
+    def test_add_error(self, bw, bad_type):
+        """Test __add__ with unsupported type raises an error"""
+        with pytest.raises(TypeError, match=f"Cannot add {type(bad_type)} to a BoseWord"):
+            bw + bad_type  # pylint: disable=pointless-statement
+
+    @pytest.mark.parametrize("bw, bad_type", tup_bw_mult_error)
+    def test_radd_error(self, bw, bad_type):
+        """Test __radd__ with unsupported type raises an error"""
+        with pytest.raises(TypeError, match=f"Cannot add {type(bad_type)} to a BoseWord"):
+            bad_type + bw  # pylint: disable=pointless-statement
+
+    @pytest.mark.parametrize("bw, bad_type", tup_bw_mult_error)
+    def test_sub_error(self, bw, bad_type):
+        """Test __sub__ with unsupported type raises an error"""
+        with pytest.raises(TypeError, match=f"Cannot subtract {type(bad_type)} from a BoseWord"):
+            bw - bad_type  # pylint: disable=pointless-statement
+
+    @pytest.mark.parametrize("bw, bad_type", tup_bw_mult_error)
+    def test_rsub_error(self, bw, bad_type):
+        """Test __rsub__ with unsupported type raises an error"""
+        with pytest.raises(TypeError, match=f"Cannot subtract a BoseWord from {type(bad_type)}"):
+            bad_type - bw  # pylint: disable=pointless-statement
+
+    WORDS_ADD = [
+        (bw1, bw2, BoseSentence({bw1: 1, bw2: 1})),
+        (bw3, bw2, BoseSentence({bw2: 1, bw3: 1})),
+        (bw2, bw2, BoseSentence({bw2: 2})),
+    ]
+
+    @pytest.mark.parametrize("f1, f2, res", WORDS_ADD)
+    def test_add_bose_words(self, f1, f2, res):
+        """Test that adding two BoseWords returns the expected BoseSentence"""
+        assert f1 + f2 == res
+        assert f2 + f1 == res
```

Comment:
This gives me an error when I run it manually in a jupyter notebook. Am I doing something wrong? :thinking: 

```
from pennylane.qchem import BoseWord

bw1 = BoseWord({(0, 0): "+", (1, 1): "-"})
bw2 = BoseWord({(0, 0): "+", (1, 0): "-"})
bw1 + bw2
```
```
TypeError: Subscripted generics cannot be used with class and instance checks
```

Reply:
Same for subtraction and multiplication by a constant.

Reply:
The code works for me @DSGuala and gives:

`BoseSentence({BoseWord({(0, 0): '+', (1, 1): '-'}): 1.0, BoseWord({(0, 0): '+', (1, 0): '-'}): 1.0})`

Reply:
Solved: use python 3.10+